sdks/python/apache_beam/io/iobase.py

```diff
@@ -886,12 +888,18 @@ def get_desired_chunk_size(total_size):
 
   def expand(self, pbegin):
     if isinstance(self.source, BoundedSource):
+      coders.registry.register_coder(BoundedSource, _MemoizingPickleCoder)
       display_data = self.source.display_data() or {}
       display_data['source'] = self.source.__class__
+
+      def output_source(_) -> BoundedSource:
+        assert isinstance(self.source, BoundedSource)
+        return self.source
+
       return (
           pbegin
           | Impulse()
-          | core.Map(lambda _: self.source)
+          | core.Map(output_source)
```

Comment:
Nit: "output_source" sounds like a noun, not a verb. We could call this "emit_source." Alternatively, you could leave this as a lambda and do `core.Map(lambda _: self.source).with_output_types(BoundedSource)`.

Reply:
done.